app/components/file_attachment_list/file_attachment_list.js

```diff
@@ -71,17 +71,14 @@ export default class FileAttachmentList extends PureComponent {
         }
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.files !== nextProps.files) {
-            this.filesForGallery = this.getFilesForGallery(nextProps);
+    componentDidUpdate(prevProps) {
+        if (prevProps.files !== this.props.files) {
+            this.filesForGallery = this.getFilesForGallery(this.props);
             this.buildGalleryFiles().then((results) => {
                 this.galleryFiles = results;
             });
         }
-        if (!this.state.loadingFiles && nextProps.files.length === 0) {
-            this.setState({
-                loadingFiles: true,
-            });
+        if (!deepEqual(this.props.files, prevProps.files) && this.props.files.length === 0) {
```

Comment:
You actually don't need to check if the files arrays are deeply equal. We store the files in a way that means that the `this.props.files`array won't change unless one of the files in it actually changes.

```suggestion
        if (this.props.files !== prevProps.files && this.props.files.length === 0) {
```